Sources/SourceKitLSP/SourceKitServer.swift

```diff
@@ -581,6 +597,14 @@ extension SourceKitServer: MessageHandler {
       default:
         reply(.failure(ResponseError.methodNotFound(R.method)))
       }
+      // We have handled the request and can't cancel it anymore.
+      // Stop keeping track of it to free the memory.
+      await self.setInProgressRequest(for: id, task: nil)
+    }
+    // Start tracking the request with a high priority to minimize the chance
+    // of cancellation
+    cancellationMessageHandlingQueue.async {
+      await self.setInProgressRequest(for: id, task: task)
     }
```

Comment:
Is this meant to be set to high priority?

Reply:
No, it’s not. Just adding the ID -> Task mapping to the dictionary isn’t really important. Once we actually cancel a request, the cancellation task (which has high priority), will be blocked on this task, which will raise its priority to `high` as well (https://developer.apple.com/wwdc23/10170?time=515). So, actually, we should run this at a very low priority because it’s just bookkeeping that doesn’t matter until you actually cancel something.

Reply:
Should the comment be changed then? :)

Reply:
Yes